Updating custom database tables tutorial for 3.x

jaredfhealy opened the pull request:
## Description

The current "storefinder" example did not work in 3.x. I updated the documentation and provided more details around the new 3.x versions. It's also updated to mention that the methodology and build requires 3.0 to work. It is not backward compatible.

## Affected versions

This is relevant to 3.x only

## Relevant issues

No issues related that I'm aware of.

Mark-H commented on en/extending-modx/tutorials/using-custom-database-tables.md:
```diff
@@ -4,256 +4,395 @@ _old_id: "330"
+**NOTE**: This was converted from the previous "StoreFinder" example.
 
-So let's say we want to create a custom component called "StoreFinder" that takes a zip code from a textfield and then looks up all the store locations with that zip code and returns them in a table. Currently we'll have one table for this: (note the prefix `modx_` - this is specific to your DB connection, done in Revolution setup.)
+**ALSO NOTE**: This methodology is not backwards compatible to MODX 2.x. This tutorial is intended to be run and used in MODX 3.
```
would be good to add a link to the 2.x docs here

jaredfhealy replied:
I noticed as well that some Markdown syntax validation failed. How do I add the link and then associate it to this pull request?

jaredfhealy replied:
Oh, I think I just push additional changes to this branch. I'll make the corrections.

jaredfhealy commented:
@Mark-H I know you're likely busy catching up after a well deserved vacation. Any further changes needed here or can we approve this PR?

Bruno17 commented:
Is there a reason, why you are using that getLoader()->addPsr4 thing in the bootstrap.php?
I've not seen that in any Extras, yet. Instead I've seen and used addPackage there.

opengeek commented:
> Is there a reason, why you are using that getLoader()->addPsr4 thing in the bootstrap.php? I've not seen that in any Extras, yet. Instead I've seen and used addPackage there.

This is so you can add other source for the extra to the PSR4 class loader. addPackage only adds the model source to it.

jaredfhealy commented:
@Bruno17 I found that originally in the 3.x docs for Namespaces. Also reviewing a few 3.x branches for things like Tagger, Collections, and GPM.
https://docs.modx.com/3.x/en/extending-modx/namespaces

Mark-H commented:
Thanks for your work on this @jaredfhealy! The updated tutorial is now live at https://docs.modx.com/3.x/en/extending-modx/tutorials/using-custom-database-tables

jaredfhealy commented:
@Mark-H Hopefully it doesn't mess it up to push back to here after you already merged it. I noticed a couple links were marked as broken on the published page because I had included ".md" in the link.

Also the formatting of the upper list was odd, so I switched that to subheadings to give it more space in between each section at the top. Also made one minor correction to the build scripts and just removed the "DIRECTORY_SEPARATOR" I was using... that crept in from somewhere else but wasn't really needed for the example.
